zenguard/zenguard.py

```diff
@@ -115,3 +131,53 @@ def pentest(self, endpoint: Endpoint, detector: Detector = None):
         scoring.score_attacks(attack_prompts)
         df = visualization.build_dataframe(attack_prompts)
         print(scoring.get_metrics(df, "Attack Instruction"))
+
+    def update_detectors(self, detectors: list[Detector]):
+        if len(detectors) == 0:
+            return {"error": "No detectors were provided"}
+
+        try:
+            response = httpx.put(
+                self._backend + "v1/detectors/update/",
+                params={"detectors": detectors},
+                headers={"x-api-key": self._api_key},
+                timeout=3,
+            )
+        except httpx.RequestError as e:
+            return {"error": str(e)}
+
+        if response.status_code != 200:
+            return {"error": str(response.json())}
+
+    def detect_in_parallel(self, prompt: str):
```

Comment:
we already have `detect` function that accepts list of detectors.

Can we alter that function to basically do 2 things:
* if detectors list has detectors then process these detectors in parallel
* if detectors list has no detectors specified then take the settings for detectors from that api key